controller/root.go

```diff
@@ -0,0 +1,121 @@
+package controller
+
+import (
+	"encoding/json"
+	"github.com/fabric8-services/fabric8-wit/app"
+	"github.com/fabric8-services/fabric8-wit/errors"
+	"github.com/fabric8-services/fabric8-wit/jsonapi"
+	"github.com/fabric8-services/fabric8-wit/log"
+	"github.com/fabric8-services/fabric8-wit/rest"
+	"github.com/fabric8-services/fabric8-wit/swagger"
+	"github.com/goadesign/goa"
+	"github.com/satori/go.uuid"
+	"net/http"
+	"strings"
+)
+
+type asseter interface {
+	Asset(name string) ([]byte, error)
+}
+
+// RootController implements the root resource.
+type RootController struct {
+	*goa.Controller
+	FileHandler asseter
+}
+
+type workingFileFetcher struct{}
+
+func (s workingFileFetcher) Asset(fileName string) ([]byte, error) {
+	return swagger.Asset(fileName)
+}
+
+var _ asseter = workingFileFetcher{}
+var _ asseter = (*workingFileFetcher)(nil)
+
+// NewRootController creates a root controller.
+func NewRootController(service *goa.Service) *RootController {
+	return &RootController{
+		Controller:  service.NewController("RootController"),
+		FileHandler: workingFileFetcher{},
+	}
+}
+
+// List runs the list action.
+func (c *RootController) List(ctx *app.ListRootContext) error {
+	roots, err := getRoot(c.FileHandler)
+
+	if err != nil {
+		return jsonapi.JSONErrorResponse(ctx, err)
+	}
+
+	res := &app.RootSingle{}
+	res.Data = convertRoot(ctx.Request, roots)
+	return ctx.OK(res)
+}
+
+// ConvertRoot converts from internal to external REST representation.
+func convertRoot(request *http.Request, root app.Root) *app.Root {
+	selfURL := rest.AbsoluteURL(request, *root.BasePath)
+	l := &app.Root{
+		Relationships: root.Relationships,
+		Attributes:    root.Attributes,
+		Links: &app.GenericLinksForRoot{
+			Self: &selfURL,
+		},
+		ID: root.ID,
+	}
+	return l
+}
+
+// Get a list of all endpoints formatted to json api format.
+func getRoot(fileHandler asseter) (app.Root, error) {
+	swaggerJSON, err := fileHandler.Asset("swagger.json")
+	if err != nil {
+		log.Error(nil, map[string]interface{}{
+			"err":  err,
+			"file": "swagger.json",
+		}, "file with id 'swagger.json' not found")
+
+		return app.Root{}, errors.NewNotFoundError("file", "swagger.json")
+	}
+
+	var result map[string]interface{}
+	json.Unmarshal([]byte(swaggerJSON), &result)
+
+	// Get and iterate over paths from swagger specification.
+	swaggerPaths := result["paths"].(map[string]interface{})
```

Comment:
I'd go the safe route by testing if a member "test" does exist and checking that the conversion did work:

```go
swaggerPathsIF, found := result["paths"]
if !found {
    // handle not found member
}
swaggerPaths, castOK := swaggerPathsIF.(map[string]interface{})
if !castOK {
    // handle cast error
}
```

If you don't do that, the core might panic/crash at run-time. In general, cast operators should always be checked.